lib/taste_tester/ssh_util.rb

```diff
@@ -15,27 +15,81 @@
 module TasteTester
   class SSH
     module Util
-      def ssh_base_cmd
-        jumps = TasteTester::Config.jumps ?
-          "-J #{TasteTester::Config.jumps}" : ''
+      def jumps
+        TasteTester::Config.jumps ? "-J #{TasteTester::Config.jumps}" : ''
+      end
+
+      def ssh_cmd_generator
+        if TasteTester::Config.ssh_cmd_gen_template
+          base_gen_args = {
+            :user => TasteTester::Config.user,
+            :jumps => jumps,
+            :host => @host,
+          }
+          TasteTester::Config.ssh_cmd_gen_template %
+            base_gen_args
+        end
+      end
+
+      def ssh_generated_cmd
+        if TasteTester::Config.ssh_cmd_gen_template
+          # we store this generated command inside a class variable
+          # so that we can directly refer to this while printing
+          # logs and error messages
+          begin
+            # run the generator command only if it's not run already
+            if @ssh_generated_cmd.nil?
+              generator = Mixlib::ShellOut.new(ssh_cmd_generator)
+              @ssh_generated_cmd = generator.run_command.stdout.chomp
+              generator.error!
```

Comment:
You can't call `.error!` on a string, and stdout returns a string. :)

```suggestion
              generator = Mixlib::ShellOut.new(ssh_cmd_generator).run_command
              generator.error!
              @ssh_generated_cmd = generator.stdout.chomp
```

Reply:
I am calling .error! on the ShellOut object right? I think it works.

I did test it by giving a bullshit arg in the template.

I will add the failure for this case in the description. I have already tested rest of the permutations which you can see in the first convo (I edited the first comment)